node-handshake.md

```diff
@@ -0,0 +1,25 @@
+# P2P node handshake
+
+_Spec v1 (2022-10-05)_
+
+Pick a publicly available P2P node (e.g. a blockchain one) implementation - which itself doesn't need to be written in Rust - and write a [network handshake](https://en.wikipedia.org/wiki/Handshaking) for it in Rust, and instructions on how to test it.
```

Comment:
> (e.g. a blockchain one)

Is handshake with any other P2P-based solution ok as well? Like some file sharing network client? Or should this be instead:

```suggestion
Pick a publicly available P2P node (more specifically: a blockchain one) implementation - which itself doesn't need to be written in Rust - and write a [network handshake](https://en.wikipedia.org/wiki/Handshaking) for it in Rust, and instructions on how to test it.
```

Reply:
I think it could be any P2P node, actually, as we're not interested in the blockchain logic specifically - just P2P capabilities.

Reply:
Example: https://ssbc.github.io/scuttlebutt-protocol-guide/#handshake.

Reply:
Or https://spec.matrix.org/v1.4/server-server-api/#joining-rooms.

Reply:
Was about to mention Matrix 😄